### ISSUE TITLE
`Undefined symbols` when integrating a package containing `FirebaseAnalytics` into a Tuist project with a target containing `FirebaseAnalytics`

### ISSUE THREAD

**Issue by TheInkedEngineer** [tuist/tuist#6256]
### What happened?

While migrating to `Tuist` we faced a problem where when we add a package containing `FirebaseAnalytics` to our main project containing a target that depends on `FirebaseAnalytics` the project will not build due to `Undefined Symbols`

### How do we reproduce it?

1. clone `https://github.com/YAtechnologies/tmp-testing-ios`
2. run tuist generate
3. build the project and you will get the errors

### Error log

Showing Recent Messages
Ld /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/DependenciesWithFirebaseIncluded.framework/DependenciesWithFirebaseIncluded normal (in target 'DependenciesWithFirebaseIncluded' from project 'TestProject')
    cd /Users/theinkedengineer/Code/Yassir/tmp/tmp-testing-ios/TestProject
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Xlinker -reproducible -target arm64-apple-ios17.4-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.4.sdk -O0 -L/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -L/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator -filelist /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/TestProject.build/Debug-iphonesimulator/DependenciesWithFirebaseIncluded.build/Objects-normal/arm64/DependenciesWithFirebaseIncluded.LinkFileList -install_name @rpath/DependenciesWithFirebaseIncluded.framework/DependenciesWithFirebaseIncluded -Xlinker -rpath -Xlinker /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/TestProject.build/Debug-iphonesimulator/DependenciesWithFirebaseIncluded.build/Objects-normal/arm64/DependenciesWithFirebaseIncluded_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/TestProject.build/Debug-iphonesimulator/DependenciesWithFirebaseIncluded.build/Objects-normal/arm64/DependenciesWithFirebaseIncluded.swiftmodule -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -framework UIKit -framework Security -lsqlite3 -lc++ -lz -framework StoreKit -lz -lsqlite3 -lc++ -lz -framework StoreKit -framework TargetContainingFirebase /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/PackageFrameworks/FirebaseAnalytics_-1C7815A27AB74526_PackageProduct.framework/FirebaseAnalytics_-1C7815A27AB74526_PackageProduct -Xlinker -no_adhoc_codesign -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/TestProject.build/Debug-iphonesimulator/DependenciesWithFirebaseIncluded.build/Objects-normal/arm64/DependenciesWithFirebaseIncluded_dependency_info.dat -o /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Products/Debug-iphonesimulator/DependenciesWithFirebaseIncluded.framework/DependenciesWithFirebaseIncluded -Xlinker -add_ast_path -Xlinker /Users/theinkedengineer/Library/Developer/Xcode/DerivedData/BugDemo-fmtgaelopldjngdbkdhcijvcdeqa/Build/Intermediates.noindex/PackageContainingFirebase.build/Debug-iphonesimulator/PackageContainingFirebase.build/Objects-normal/arm64/PackageContainingFirebase.swiftmodule

Undefined symbols for architecture arm64:
  "_GULIsLoggableLevel", referenced from:
      -[APMMonitor isLoggableLevel:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
  "_GULLogBasic", referenced from:
      -[APMASLLogger logMessage:logTag:messageCode:withLogLevel:] in GoogleAppMeasurement[arm64][12](APMASLLogger.o)
  "_GULLogError", referenced from:
      -[APMPersistentDictionary initWithFileName:] in GoogleAppMeasurement[arm64][257](APMPersistentDictionary.o)
      _APMWriteDictionaryToURL in GoogleAppMeasurement[arm64][257](APMPersistentDictionary.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][311](APMUserDefaults.o)
  "_GULLogInfo", referenced from:
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][330](UIViewController+APMScreenClassName.o)
  "_GULLogWarning", referenced from:
      +[APMMeasurement sharedInstance] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMPersistentDictionary objectForKey:] in GoogleAppMeasurement[arm64][257](APMPersistentDictionary.o)
      -[APMPersistentDictionary setObject:forKey:] in GoogleAppMeasurement[arm64][257](APMPersistentDictionary.o)
      -[APMUserDefaults objectForKey:] in GoogleAppMeasurement[arm64][311](APMUserDefaults.o)
      -[APMUserDefaults setObject:forKey:] in GoogleAppMeasurement[arm64][311](APMUserDefaults.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][311](APMUserDefaults.o)
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][330](UIViewController+APMScreenClassName.o)
      ...
  "_GULSetLoggerLevel", referenced from:
      -[APMMonitor setDebugModeEnabled:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
      -[APMMonitor setDebugModeEnabled:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
      -[APMMonitor setVerboseLoggingEnabled:] in GoogleAppMeasurement[arm64][68](APMMonitor.o)
  "_OBJC_CLASS_$_GULAppDelegateSwizzler", referenced from:
       in GoogleAppMeasurement[arm64][7](APMAnalytics.o)
  "_OBJC_CLASS_$_GULAppEnvironmentUtil", referenced from:
       in GoogleAppMeasurement[arm64][5](APMAEU.o)
  "_OBJC_CLASS_$_GULMutableDictionary", referenced from:
       in GoogleAppMeasurement[arm64][35](APMEExperiment.o)
  "_OBJC_CLASS_$_GULNetwork", referenced from:
       in GoogleAppMeasurement[arm64][7](APMAnalytics.o)
  "_OBJC_CLASS_$_GULSwizzler", referenced from:
       in GoogleAppMeasurement[arm64][330](UIViewController+APMScreenClassName.o)
  "_kGULNetworkHTTPStatusCodeCannotAcceptTraffic", referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
  "_kGULNetworkHTTPStatusCodeNotFound", referenced from:
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
  "_kGULNetworkHTTPStatusCodeNotModified", referenced from:
      ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement(SKAN) handleSKANFetchResponse:data:error:] in GoogleAppMeasurement[arm64][66](APMMeasurement+SKAN.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][67](APMMeasurement+URL.o)
  "_kGULNetworkHTTPStatusCodeUnavailable", referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
  "_kGULNetworkHTTPStatusNoContent", referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][67](APMMeasurement+URL.o)
  "_kGULNetworkHTTPStatusOK", referenced from:
      -[APMETaskManager handleFetchingExperimentsResponse:data:error:] in GoogleAppMeasurement[arm64][42](APMETaskManager.o)
      -[APMMeasurement networkUploadCompletionHandlerWithResponse:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      ___37-[APMMeasurement retrievePSMResponse]_block_invoke in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement networkRemoteConfigFetchCompletionHandler:data:error:] in GoogleAppMeasurement[arm64][62](APMMeasurement.o)
      -[APMMeasurement(SKAN) handleSKANFetchResponse:data:error:] in GoogleAppMeasurement[arm64][66](APMMeasurement+SKAN.o)
      ___46-[APMMeasurement(URL) requestDeferredDeepLink]_block_invoke in GoogleAppMeasurement[arm64][67](APMMeasurement+URL.o)
  "_pb_decode", referenced from:
      _apmpb_decode_app_property in GoogleAppMeasurement[arm64][78](apmpb_app_property_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][81](apmpb_audience_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][81](apmpb_audience_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][84](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][84](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][87](apmpb_client_property_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][87](apmpb_client_property_decoder.o)
      ...
  "_pb_decode_varint", referenced from:
      _nano_decode_repeated_uint64 in GoogleAppMeasurement[arm64][315](decoder_common.o)
  "_pb_encode", referenced from:
      _nano_two_pass_encoding in GoogleAppMeasurement[arm64][316](encoder_common.o)
      _nano_two_pass_encoding in GoogleAppMeasurement[arm64][316](encoder_common.o)
  "_pb_encode_string", referenced from:
      _nano_encode_string in GoogleAppMeasurement[arm64][316](encoder_common.o)
  "_pb_encode_submessage", referenced from:
      _apmpb_encode_repeated_app_property in GoogleAppMeasurement[arm64][79](apmpb_app_property_encoder.o)
      _apmpb_encode_repeated_audience in GoogleAppMeasurement[arm64][82](apmpb_audience_encoder.o)
      _apmpb_encode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][85](apmpb_audience_leaf_filter_result_encoder.o)
      _apmpb_encode_repeated_client_property in GoogleAppMeasurement[arm64][88](apmpb_client_property_encoder.o)
      _apmpb_encode_repeated_consent_config in GoogleAppMeasurement[arm64][91](apmpb_consent_config_encoder.o)
      _apmpb_encode_repeated_consent_default_entry in GoogleAppMeasurement[arm64][94](apmpb_consent_default_entry_encoder.o)
      _apmpb_encode_repeated_consent_delegation_entry in GoogleAppMeasurement[arm64][97](apmpb_consent_delegation_entry_encoder.o)
      ...
  "_pb_encode_tag_for_field", referenced from:
      _apmpb_encode_repeated_app_property in GoogleAppMeasurement[arm64][79](apmpb_app_property_encoder.o)
      _apmpb_encode_repeated_audience in GoogleAppMeasurement[arm64][82](apmpb_audience_encoder.o)
      _apmpb_encode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][85](apmpb_audience_leaf_filter_result_encoder.o)
      _apmpb_encode_repeated_client_property in GoogleAppMeasurement[arm64][88](apmpb_client_property_encoder.o)
      _apmpb_encode_repeated_consent_config in GoogleAppMeasurement[arm64][91](apmpb_consent_config_encoder.o)
      _apmpb_encode_repeated_consent_default_entry in GoogleAppMeasurement[arm64][94](apmpb_consent_default_entry_encoder.o)
      _apmpb_encode_repeated_consent_delegation_entry in GoogleAppMeasurement[arm64][97](apmpb_consent_delegation_entry_encoder.o)
      ...
  "_pb_encode_varint", referenced from:
      _nano_encode_repeated_uint64 in GoogleAppMeasurement[arm64][316](encoder_common.o)
  "_pb_istream_from_buffer", referenced from:
      _apmpb_decode_app_property in GoogleAppMeasurement[arm64][78](apmpb_app_property_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][81](apmpb_audience_decoder.o)
      _apmpb_decode_audience in GoogleAppMeasurement[arm64][81](apmpb_audience_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][84](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_audience_leaf_filter_result in GoogleAppMeasurement[arm64][84](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][87](apmpb_client_property_decoder.o)
      _apmpb_decode_client_property in GoogleAppMeasurement[arm64][87](apmpb_client_property_decoder.o)
      ...
  "_pb_ostream_from_buffer", referenced from:
      _nano_two_pass_encoding in GoogleAppMeasurement[arm64][316](encoder_common.o)
  "_pb_read", referenced from:
      _apmpb_decode_repeated_app_property in GoogleAppMeasurement[arm64][78](apmpb_app_property_decoder.o)
      _apmpb_decode_repeated_app_property in GoogleAppMeasurement[arm64][78](apmpb_app_property_decoder.o)
      _apmpb_decode_repeated_audience in GoogleAppMeasurement[arm64][81](apmpb_audience_decoder.o)
      _apmpb_decode_repeated_audience in GoogleAppMeasurement[arm64][81](apmpb_audience_decoder.o)
      _apmpb_decode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][84](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_repeated_audience_leaf_filter_result in GoogleAppMeasurement[arm64][84](apmpb_audience_leaf_filter_result_decoder.o)
      _apmpb_decode_repeated_client_property in GoogleAppMeasurement[arm64][87](apmpb_client_property_decoder.o)
      _apmpb_decode_repeated_client_property in GoogleAppMeasurement[arm64][87](apmpb_client_property_decoder.o)
      ...
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Undefined symbol: _GULIsLoggableLevel

Undefined symbol: _GULLogBasic

Undefined symbol: _GULLogError

Undefined symbol: _GULLogInfo

Undefined symbol: _GULLogWarning

Undefined symbol: _GULSetLoggerLevel

Undefined symbol: _OBJC_CLASS_$_GULAppDelegateSwizzler

Undefined symbol: _OBJC_CLASS_$_GULAppEnvironmentUtil

Undefined symbol: _OBJC_CLASS_$_GULMutableDictionary

Undefined symbol: _OBJC_CLASS_$_GULNetwork

Undefined symbol: _OBJC_CLASS_$_GULSwizzler

Undefined symbol: _kGULNetworkHTTPStatusCodeCannotAcceptTraffic

Undefined symbol: _kGULNetworkHTTPStatusCodeNotFound

Undefined symbol: _kGULNetworkHTTPStatusCodeNotModified

Undefined symbol: _kGULNetworkHTTPStatusCodeUnavailable

Undefined symbol: _kGULNetworkHTTPStatusNoContent

Undefined symbol: _kGULNetworkHTTPStatusOK

Undefined symbol: _pb_decode

Undefined symbol: _pb_decode_varint

Undefined symbol: _pb_encode

Undefined symbol: _pb_encode_string

Undefined symbol: _pb_encode_submessage

Undefined symbol: _pb_encode_tag_for_field

Undefined symbol: _pb_encode_varint

Undefined symbol: _pb_istream_from_buffer

Undefined symbol: _pb_ostream_from_buffer

Undefined symbol: _pb_read

Linker command failed with exit code 1 (use -v to see invocation)

### macOS version

14.4.1

### Tuist version

4.11.0

### Xcode version

15.3

**Comment by danibachar** [comment]
You are consuming the internal swift package as a tuist project. This is not the way to consume swift packages with Tuist. 
I suggest on following [this guide](https://docs.tuist.io/guide/project/dependencies.html#external-dependencies) on how to consume external dependencies. 

You can also look in the [fixtures folder](https://github.com/tuist/tuist/tree/main/fixtures/app_with_spm_dependencies) for examples on how to setup a project that is structured like yours.

**Comment by TheInkedEngineer** [comment]
@danibachar that was just an example, in the actual project, the package in an external dependency. that being said, I will check those links out and get back to you later today, thanks.

**Comment by MouadBenjrinija** [comment]
> You are consuming the internal swift package as a tuist project. This is not the way to consume swift packages with Tuist. I suggest on following [this guide](https://docs.tuist.io/guide/project/dependencies.html#external-dependencies) on how to consume external dependencies.
> 
> You can also look in the [fixtures folder](https://github.com/tuist/tuist/tree/main/fixtures/app_with_spm_dependencies) for examples on how to setup a project that is structured like yours.

@danibachar 
If the suggested approach is to use the  [Tuist's XcodeProj-based integration](https://docs.tuist.io/guide/project/dependencies#tuist-s-xcodeproj-based-integration) then it doesn't work for all external packages.. as some of them is not compatible with this method, (e.g. [XMPPFramework](https://github.com/robbiehanson/XMPPFramework) , previously reported by someone [here](https://github.com/tuist/tuist/issues/5195))
So only [Xcode's default integration](https://docs.tuist.io/guide/project/dependencies#xcode-s-default-integration) works for these packages.

**Comment by pepicrft** [comment]
I debugged this issue, and the problem is unrelated to Tuist. [The fixture project](https://github.com/YAtechnologies/tmp-testing-ios) is duplicated the symbols of `FirebaseAnalytics` (transitive static product) across various dynamic frameworks. I suggested as a proper setup:

- Either use Tuist's default integration using Xcode primitives and changing the type of Firebase products to be dynamic.
- Wrap Firebase static targets into a dynamic framework and use `@_exported import FirebaseAnalytics` so that symbols are only in a single dynamic framework.
